corehq/apps/app_manager/views/download.py

```diff
@@ -368,12 +383,36 @@ def validate_form_for_build(request, domain, app_id, unique_form_id, ajax=True):
         return HttpResponse(response_html)
 
 
-def download_index_files(app):
+def _file_needed_for_CCZ(path, prefix, profiles):
+    if path.startswith(prefix):
+        # already narrowed to specific profile
+        if prefix != 'files/':
+            return True
+        else:
+            second = path.split('/')[1]
+            # dont want to return files from other profiles
+            if second in profiles:
+                return False
+            else:
+                return True
+    else:
+        return False
+
+
+def download_index_files(app, build_profile_id=None):
     files = []
+    prefix = 'files/'
+    profiles = set(app.build_profiles.keys())
+    if build_profile_id:
+        prefix += build_profile_id + '/'
     if app.copy_of:
-        files = [(path[len('files/'):], app.fetch_attachment(path))
+        # profile hasnt been built yet
+        if not (prefix + 'profile.ccpr') in app._attachments:
+            app.create_build_files(save=True, build_profile_id=build_profile_id)
+            app.save()
+        files = [(path[len(prefix):], app.fetch_attachment(path))
                  for path in app._attachments
-                 if path.startswith('files/')]
+                 if _file_needed_for_CCZ(path, prefix, profiles)]
     else:
         files = app.create_all_files().items()
```

Comment:
Is it a concern that this can return paths that contain the build profile id? Asking because the true branch of this if-block always strips the prefix.